src/lib/structures/symbols.ml

```diff
@@ -367,18 +367,61 @@ let print_clean fmt s = Format.fprintf fmt "%s" (to_string_clean s)
 let print fmt s = Format.fprintf fmt "%s" (to_string s)
 
 
-let fresh, reinit_fresh_sy_cpt =
-  let cpt = ref 0 in
-  let fresh ?(is_var=false) s =
-    incr cpt;
-    (* garder le suffixe "__" car cela influence l'ordre *)
-    let s = (Format.sprintf "!?__%s%i" s (!cpt)) in
-    if is_var then var @@ Var.of_string s else name s
-  in
-  let reinit_fresh_sy_cpt () =
-    cpt := 0
-  in
-  fresh, reinit_fresh_sy_cpt
+module type Id = sig
+  val fresh : ?base:string -> unit -> string
+  val reset_fresh_cpt : unit -> unit
+  val is_id : string -> bool
+  val make_as_fresh : string -> string
+end
+
+module MakeId(S : sig val prefix : string end) : Id = struct
+
+  let make_as_fresh = (^) S.prefix
+
+  let fresh, reset_fresh_cpt =
+    let cpt = ref 0 in
+    let fresh_string ?(base = "") () =
+      incr cpt;
+      make_as_fresh (base ^ (string_of_int !cpt))
+    in
+    let reset_fresh_string_cpt () =
+      cpt := 0
+    in
+    fresh_string, reset_fresh_string_cpt
+
+  let len_pre = String.length S.prefix
+
+  let is_id s =
+    let len_s = String.length s in
+    let rec aux i =
+      if i = len_pre then true
+      else
+        Char.equal (String.unsafe_get s i) (String.unsafe_get S.prefix i)
+        && aux (i + 1)
+    in len_s >= len_pre && aux 0
```

Comment:
```suggestion
    in len_s > len_pre && aux 0
```
Assuming that names as empty strings don't occur

Reply:
`||` is a valid SMTLib2 identifier corresponding to the empty string.

Reply:
Oh gosh... the SMT-LIB accepts too much things as identifiers...